### ISSUE TITLE
Inconsistency with the paper

### ISSUE THREAD

**Issue by razvanc92** [Luoyonghong/E2EGAN#4]
Hello, firstly I would like to thank you for sharing the code but I have noticed some inconsistencies with the paper which I would like to clarify.

1. In E2EGAN_PHY.py the discriminator loss function is not the same as the one presented in the paper (line 381). Also why is the discriminator loss dependent on the l2_loss, from my understanding the discriminator optimizer can not influence that part of the loss.
2. In gru_forGAN.py line 110 you do not have batch norm which in the paper you said that you used, or it is applied somewhere else?

In addition I'm having some questions regarding the intuition behind some of the decisions which are not touch on the paper.
3. I've noticed that you clipped all your parameters in both the generator and discriminator. Looking at WGAN original implementation where they only clipped the discriminator, why would you need to clip the generator as well?

Looking forward for you answer.
